### DISCUSSION TITLE
Field References

### DISCUSSION THREAD

**Discussion by shk3bq4d** [dani-garcia/vaultwarden#2320] [Ideas]
Hello,

I came here to duplicate a (I believe unfilled) feature requests listed on community.bitwarden.com

I'm not sure to what extents its appropriate or has already been discussed elsewhere. Apologies for duplicates. 

https://community.bitwarden.com/t/field-references/1624

There is a also a reference from the keepass implementation of the feature: https://keepass.info/help/base/fieldrefs.html

And I am cloning the content of the comment that illustrates, I believe, the most the actual need: https://community.bitwarden.com/t/field-references/1624/3

from: philipsd6 Philip Feb '19

> I have many accounts that use my domain username and password, but the username has to be formatted differently sometimes, i.e.
> 
> philipsd6@example.com
> DOMAIN\philipsd6
> full.name
> full.name@example.com
> DEVDOMAIN\prefix-philipsd6
> 
> In all cases, the password is the same, and it expires every 6 weeks, so the entries must be kept in sync. Without field references, this is difficult to manage.

**Comment by BlackDex** [comment]
I'm not sure what you expect from the maintainers from Vaultwarden. But this really looks more like something for the Bitwarden clients, which is not maintained by this project.

**Comment by shk3bq4d** [comment]
Hello and thank you for your answer,

I lack the technical understanding of how clients, backend GUI and backend work altogether and I also lack the intended limitation scope of the Vaultwarden project.

With my current limited understanding, I have the feeling the feature could, in theory, be implemented either totally client side or totally backend side (if the backend GUI supports it), its usability may obviously vary depending on how it's done.

But I might be wrong and I am happy to hear as such and am thankful for all the work you have committed so far.

I still see it as important that at least this idea was raised in here but we can stop the discussion at this point.

Have a good day!

**Reply by BlackDex** [reply]
The backend/server API doesn't know what the values are, so i can't do any formatting what so ever.
It only has encrypted values, so it can't even match a site. The backend also doesn't do any decryption since it doesn't have the password.
So, something like this needs to be done on the client side.
